File: examples/module3/vararg/Vararg.java
public class Vararg {

    static void strlist(String... strings) {
        for (String s : strings) {
            System.out.println(s);
        }
    }

    public static void main(String[] args) {
        strlist("string1", "string2", "string3");
        strlist("string1", "string3");
        
        //strlist("string1", "string3", 1234);
    }
}

File: examples/module3/vararg/vararg.c
#include <stdio.h>
#include <stdarg.h>

void strlist(int n, ...) {
    va_list ap;
    char *s;
    va_start(ap, n);
    while (1)
    {
        s = va_arg(ap, char *);
        printf("%s\n", s);
        n--;
        if (n == 0)
            break;
    }
    va_end(ap);
}


int main(void) {
    strlist(3, "string1", "string2", "string3");
    strlist(2, "string1", "string3");

    //strlist(3, "string1", "string2", 1234);
}